### PR TITLE
Unify local kind

### PR DESCRIPTION
**Pull request by michalbiesek**
- [x] Depends from #457

---
This change is [Reviewable](https://reviewable.io/reviews/memkind/memkind/462)

### CODE REVIEW

**Comment by michalbiesek** [comment]
The patch is pretty big so I recommend to review this commit by commit

**Review by PatKamin — APPROVED** [review]

**Review by bratpiorka — COMMENTED** [review]
> Reviewed 12 of 17 files at r1, 10 of 10 files at r2, 1 of 3 files at r3.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/462)* status: all files reviewed, 1 unresolved discussion (waiting on @michalbiesek and @PatKamin)

___
*[src/memkind_mem_attributes.c, line 172 at r3](https://reviewable.io/reviews/memkind/memkind/462#-MR52ypp1pMPJu-997ly:-MR52ypp1pMPJu-997lz:b-6wmywa) ([raw file](https://github.com/memkind/memkind/blob/d4e880404398dc504891a5bd2269159daae97bac/src/memkind_mem_attributes.c#L172)):*
> ```C
>             numa_bitmask_weight(attr_loc_mask) > 1) {
>             ret = MEMKIND_ERROR_MEMTYPE_NOT_AVAILABLE;
>             log_err("Multiple NUMA Nodes have the same highest local capacity for init node %d.",
> ```

"same highest local capacity" - this is ok?

**Review by michalbiesek — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/462)* status: 9 of 10 files reviewed, 1 unresolved discussion (waiting on @bratpiorka and @PatKamin)

___
*[src/memkind_mem_attributes.c, line 172 at r3](https://reviewable.io/reviews/memkind/memkind/462#-MR52ypp1pMPJu-997ly:-MR588B9CMMTrimOIoku:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/d4e880404398dc504891a5bd2269159daae97bac/src/memkind_mem_attributes.c#L172)):*
<details><summary><i>Previously, bratpiorka (Rafał Rudnicki) wrote…</i></summary><blockquote>

"same highest local capacity" - this is ok?
</blockquote></details>

Done.

File: src/memkind_local.c [hunk 1]
```diff
@@ -6,26 +6,41 @@
...
 };
 
-static struct hi_cap_loc_numanodes_t
-    memkind_hi_cap_loc_numanodes_g[NODE_VARIANT_MAX];
+static inline void local_kind_finalize(const struct loc_numanodes_t *g)
+{
+    if (MEMKIND_LIKELY(!g->init_err)) {
+        if (g->per_cpu_numa_nodes) {
+            int num_cpus = numa_num_configured_cpus();
+            int i;
```
Comment (PatKamin) [inline src/memkind_local.c:19]: Move this definition into the loop, you're not using this variable anywhere else.

**Review by michalbiesek — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/462)* status: 7 of 10 files reviewed, 2 unresolved discussions (waiting on @bratpiorka and @PatKamin)

___
*[src/memkind_local.c, line 19 at r4](https://reviewable.io/reviews/memkind/memkind/462#-MR3w4Y0wgChGAkhgbF6-r4-19:-MR5_6xz6XCxNad0UT0t:b-h4390b) ([raw file](https://github.com/memkind/memkind/blob/757f013edc9c412d9b6c5b8db2d50db0d91c0f7a/src/memkind_local.c#L19)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

Move this definition into the loop, you're not using this variable anywhere else.
</blockquote></details>

See that we support in Travis GCC4.8 and compiler in this version required this, see: https://travis-ci.org/github/memkind/memkind/jobs/748134469

**Review by bratpiorka — COMMENTED** [review]
> Reviewed 2 of 3 files at r5, 1 of 1 files at r6.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/462)* status: all files reviewed, 2 unresolved discussions (waiting on @bratpiorka and @PatKamin)

**Pull request merged**